surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,360 @@
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use with Rust?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`
+- Stack traces
+- Traditional debuggers
+- Record/replay debuggers
+- High-level tracing (`dtrace`/`ebpf`/`ETW`)
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What uses of Rust are you debugging?
+Type: select all that apply (optional)
+
+- Single-threaded application
+- Multi-threaded application
+- Pure Rust application
+- Mixed C or C++ and Rust application
+- Crash dumps
+- Cloud/server-side application
+- Embedded hardware
+
+> *Justification:*
+>
+> Some debugger use cases are very different from "standard" debugging.
+
+### Do you debug Rust remotely?
+i.e. are you debugging something that's running on an embedded device/another
+platform, or something like that
+
+Type: select all that apply
+
+- Yes, debugger on the host (e.g. `gdbserver` or wired connection to embedded device)
+- Yes, debugger on the device (e.g. an SSH session with a debugger running)
+- No
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging Rust?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Learning how software works
+- Understand runtime semantics
+- Understand a third-party library's behaviour
+- Understand own program's behaviour
+- Profiling
+- Investigating a crash dump
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
+If: uncomfortable with debugger
+Type: select all that apply
+
+- Unfamiliar (just need to spend more time)
+- Reliability (don't expect debugger to work well enough)
+- Ease-of-use
+- Lack of documentation with Rust
+
+> *Justification:*
+>
+> Why aren't users comfortable with debuggers is useful.
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you use a debugger with?
```

Comment:
Should we ask which they use "regularly". Otherwise, they might be tempted to answer languages they've used debuggers for on rare occasion which we don't really care about.

Reply:
Added "regularly".